app/src/main/java/com/hedvig/app/feature/loggedin/ui/LoggedInViewModel.kt

```diff
@@ -21,6 +24,8 @@ abstract class LoggedInViewModel : ViewModel() {
     private val _toolbarInset = MutableLiveData<Int>()
     val toolbarInset: LiveData<Int> = _toolbarInset
 
+    abstract val shouldOpenReviewDialog: LiveData<Boolean>
```

Comment:
This thing looks like it may be representing an event, rather than data, right? 🤔 There's a tricky problem with `LiveData` and events, in that any data represented by `LiveData` needs to be repeatable on configuration change. If not, some unpleasant bugs can occur. Imagine the following scenario:

1. We publish `true` on this `LiveData`
2. The user gets the rating dialog
3. The user dismisses the dialog
4. The user rotates the screen/resizes the window/changes the device theme
5. The dialog re-appears, because the `LiveData` re-triggers post-configuration change

The solution I've employed most times to remedy this issue is to use `SharedFlow`, which is more appropriate for events (that should not repeat after configuration change).

Reply:
Additionally, I've been favoring `StateFlow` and `SharedFlow` as a replacement entirely for `LiveData` - iirc, Google now recommends these over `LiveData` for Kotlin-codebases 😉

Reply:
Finally, should we make this non-abstract? I've been favoring the producer-consumer pairs of mutable/non-mutable in the abstract class, as we generally never put anything but an underlying `Mutable{StateFlow|SharedFlow|LiveData}` in there at the end of the day 🤷

Reply:
Fixed, thanks for the comments!